### PR TITLE
Use lower quorum for smaller validator sets

### PR DESCRIPTION
**Pull request by wilsonianb**

### CODE REVIEW

File: src/ripple/app/misc/ValidatorList.h [hunk 1]
```diff
@@ -412,11 +415,12 @@ ValidatorList::onConsensusStart (
 
     auto size = rankedKeys.size();
 
-    // Use all eligible keys if there is only one trusted list.
-    if (publisherLists_.size() == 1)
+    // Use all eligible keys if there is less than 10 listed validators or
+    // only one trusted list
+    if (size < 10 || publisherLists_.size() == 1)
     {
         // Raise the quorum to 80% of the trusted set
```
Comment (bachase) [inline src/ripple/app/misc/ValidatorList.h]: Update this comment too since not always 80%?

File: src/test/app/ValidatorList_test.cpp [hunk 2]
```diff
@@ -733,13 +748,72 @@ class ValidatorList_test : public beast::unit_test::suite
...
+
+            auto const localKey = randomNode();
+            std::vector<std::string> cfgPublishers;
+            hash_set<PublicKey> activeValidators;
+
+            std::vector<std::string> cfgKeys {
+                toBase58(TokenType::TOKEN_NODE_PUBLIC, localKey)};
+            cfgKeys.reserve(9);
+
+            constexpr std::array<std::size_t, 10> quorums = { 0, 1, 2, 3, 3, 4, 4, 5, 6, 7 };
```
Comment (bachase) [inline src/test/app/ValidatorList_test.cpp]: Consider a comment here too indicating that this is the +1 buffer for unlisted local key?

**Comment by JoelKatz** [comment]
:neckbeard::+1:

**Comment by seelabs** [comment]
In 0.60.0-rc3

**Pull request closed without merging**